cadc-gms/src/main/java/org/opencadc/auth/PosixMapperClient.java

```diff
@@ -0,0 +1,150 @@
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.auth;
+
+import ca.nrc.cadc.auth.PosixPrincipal;
+import java.net.URI;
+import java.util.Iterator;
+import java.util.List;
+import javax.security.auth.Subject;
+import org.apache.log4j.Logger;
+import org.opencadc.gms.GroupURI;
+
+/**
+ * Stub to figure out API.
+ * 
+ * @author pdowler
+ */
+public class PosixMapperClient {
+    private static final Logger log = Logger.getLogger(PosixMapperClient.class);
+
+    private final URI resourceID;
+    
+    public PosixMapperClient(URI resourceID) {
+        this.resourceID = resourceID;
+    }
+    
+    // use case: cavern does this when caller tried to set group permissions
+    // if we have multiple, this is convenience
+    // intent: this may create and persist a local GID as a side effect
+    public Integer getGID(GroupURI guri) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // or check multiple in one call??
+    public List<PosixGroup> getGID(List<GroupURI> groups) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: cavern does this when reading a node from disk and output the node doc
+    // if we have multiple, this is convenience
+    public GroupURI getURI(Integer gid) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // or check multiple in one call??
+    public List<PosixGroup> getURI(List<GroupURI> groups) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: cavern needs PosixPrincipal in the subject
+    // use case: StandardIdentityManager calls this to add local identity
+    public void augment(Subject s) {
+        throw new UnsupportedOperationException();
+    }
+    
+    // use case: skaha needs to complete username-uid map for user containers
+    public Iterator<PosixPrincipal> getUserMap() {
```

Comment:
The `PosixPrincipal` only contains a UID.  For a complete mapping, this would need to include the Name, Home Directory, and default Shell, which could just default to `/sbin/nologin`.  Is that the purpose of this method?

Reply:
I'm adding username and defaultGroup to PosixPrincipal (cavern only needs the default group). In principle posix username is local and thus could differ from "network" username (HttpPrincipal) so I'm pretty sure it should be there but we don't know enough about the possibilities there.

yes, the purpose is to provide a local uid<->username mapping to construct a "passwd" file... I'm hoping minimal user info can reside solely in the Subject and that looks plausible, but we might need to add more fields to `PosixPrincipal`

Reply:
see https://github.com/opencadc/core/pull/230 for the PosixPrincipal change (so far)

see other changes in this PR to make the AC augment subject calls get those posix values to the client (build failing until cadc-util is published)